### DISCUSSION TITLE
Prometheus metrics - gauge type for time_taken

### DISCUSSION THREAD

**Discussion by aizerin** [jertel/elastalert2#986] [Ideas]
I'm trying to use Prometheus metric from elastalert2 but I'm not able to visualize individual time consumption for rule in time. Only thing that is available is the total time taken like a counter. 

Like this: 

<img width="916" alt="image" src="https://user-images.githubusercontent.com/18553162/197519613-9ae4f2c4-d4e0-462e-a309-0bbf3eae89e6.png">

But It would be more useful for rule debugging to visualize individual runs of rules. Like in this issue https://github.com/Yelp/elastalert/issues/2026

<img width="916" alt="image" src="https://user-images.githubusercontent.com/18553162/197519850-66c6f0eb-df1b-42d2-999c-5dddaa5ac4a3.png">

Or is there a way how to make it?

**Comment by jertel** [comment]
If Grafana can do a diff between data point entries then it would have access to the new runtime delta for the latest run time. 

If you wanted to add a separate datapoint to the prometheus set you could submit a PR. The code is in the `elastalert/prometheus_wrapper.py` file.
